src/main/java/seedu/address/model/country/CountryManager.java

```diff
@@ -19,23 +23,63 @@ public CountryManager() {
         countryCodeMap = initCountryCodeMap();
     }
 
+    /**
+     * Checks if countryCode is a valid ISO3166 code.
+     *
+     * @param countryCode The country code.
+     * @return Whether countryCode is a valid ISO3166 code.
+     */
+    public static boolean isValidCode(String countryCode) {
+        for (int i = 0; i < COUNTRY_CODES.length; i++) {
+            if (COUNTRY_CODES[i].equals(countryCode)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     private static Map<String, Country> initCountryCodeMap() {
         Map<String, Country> countryCodeMap = new HashMap<>();
-        String[] countryCodes = Locale.getISOCountries();
-        for (String countryCode : countryCodes) {
+        for (String countryCode : COUNTRY_CODES) {
             countryCodeMap.put(countryCode, new Country(countryCode));
         }
         return countryCodeMap;
     }
 
+    /**
+     * Checks if {@code country} contains {@code countryNote}.
+     *
+     * @param country     The country to be checked.
+     * @param countryNote The note to be checked.
+     * @return Whether {@code country} contains {@code countryNote}.
+     */
+    public boolean hasCountryNote(Country country, Note countryNote) {
+        if (!isValidCode(country.getCountryCode())) {
+            return false;
+        }
+        return countryCodeMap.get(country.getCountryCode()).hasCountryNote(countryNote);
+    }
```

Comment:
Hmmm I was thinking that this CountryManager class holds all Country objects, while every other class should be holding references to these objects instead of any new Country() object. That way, the class themselves should be able to check if the Country has any notes, and this method would not be needed.

Reply:
My idea is that whenever you are interacting with the existing `Country` instances (other than just getting country notes), you should only do it via `CountryManager`. I think this layer of abstraction is good especially for logging/debugging in the future since `Model` only needs to operate on `CountryManager` and I would be able to log all `Country` method calls.

For classes like the `CountryNoteParser` or `CountryFilterParser` they should not be holding references to the existing Country as they do not have a reference to `CountryManager`. Only `Model` has a reference to `CountryManager`. So when a `Command` executes on a `Model`, then they are able to check for duplicate notes, etc.

Also, `Country` has a `hasCountryNote` method, but I purposely made it `protected` so that only `CountryManager` can call that method. Other classes should only be able to call the constructor and `getCountryNote` method of `Country` only.